### ISSUE TITLE
Azure DevOps Release step with "Cosmos DB Data Migration" fails

### ISSUE THREAD

**Issue by cloud20x** [Winvision/vsts-tasks#37]
**Describe the bug**
When running an azure devops release, a step with "Cosmos DB Data Migration" fails with the following error message:
```
2023-05-26T12:09:16.7410463Z (node:6476) Warning: Use Cipheriv for counter mode of aes-256-ctr
2023-05-26T12:09:18.6551975Z ERR:Unexpected HTTP response: 404
2023-05-26T12:09:18.6569867Z ##[error]Error: Parameter 'toolPath' cannot be null or empty.
```

digging a bit deeper, it seems, that the download is not reachable anymore:
Downloading: https://download.microsoft.com/download/E/1/4/E143A339-41AE-4E0E-9CC8-911C0B663478/dt-1.7.zip

(it did work until end of 2022)

**To Reproduce**
1. Create an azure devops release
2. create a step with "Cosmos DB Data Migration" 
3. run the release

**Expected behavior**
It should be possible to download the tool and run the "Cosmos DB Data Migration" step

**Screenshots**
<img width="1724" alt="image" src="https://github.com/Winvision/vsts-tasks/assets/98384112/7dab2320-8990-4747-b05c-6163897d957f">

**Desktop (please complete the following information):**
 - Azure DevOps
 - Browser: Chrome
 - Version: latest

**Comment by Nimtz89** [comment]
@cloud20x I see you're running version 1.0.13, can you update to the latest version (1.0.17) and check if the issue persists.

**Comment by cloud20x** [comment]
@Nimtz89 Thanks a lot for the hint, with the new version, I get another error message:
```
File not found: 'D:\a\_tasks\CosmosDBCollectionManagement_dc86e5f9-18e2-492b-8d35-588e526aa3b1\1.0.16\index.js' 
```
Do you have any idea why?

**Comment by Nimtz89** [comment]
@cloud20x The task.json files weren't updated, this will be done in 1.0.18

**Comment by cloud20x** [comment]
@Nimtz89 Thanks for the info. Is there a plan when this may happen? We use the "Cosmos DB tasks" in our load process that is currently "broken" due to the reported issues :-(

**Comment by Nimtz89** [comment]
@cloud20x The update to 1.0.18 just finished

**Comment by cloud20x** [comment]
@Nimtz89 sadly, the version 1.0.18 didn't resolve the issue. I still get:
```
##[error]File not found: 'D:\a\_tasks\CosmosDBCollectionManagement_dc86e5f9-18e2-492b-8d35-588e526aa3b1\1.0.18\index.js'
```
am I doing something wrong?
Thanks a lot for your support

**Comment by Nimtz89** [comment]
@cloud20x On what kind of agent are you running the job?

**Comment by cloud20x** [comment]
@Nimtz89 our job is running on a "hosted agent" in version "3.220.2" called "windows-latest":
```
2023-05-30T11:20:57.6525424Z ##[section]Starting: Initialize job
2023-05-30T11:20:57.6532210Z Agent name: 'Hosted Agent'
2023-05-30T11:20:57.6532851Z Agent machine name: 'WIN-4HRJR9TKEKI'
2023-05-30T11:20:57.6533120Z Current agent version: '3.220.2'
2023-05-30T11:20:57.6630966Z ##[group]Operating System
2023-05-30T11:20:57.6631287Z Microsoft Windows Server 2022
2023-05-30T11:20:57.6631439Z 10.0.20348
2023-05-30T11:20:57.6631549Z Datacenter
2023-05-30T11:20:57.6631677Z ##[endgroup]
2023-05-30T11:20:57.6631785Z ##[group]Runner Image
2023-05-30T11:20:57.6631926Z Image: windows-2022
2023-05-30T11:20:57.6632058Z Version: 20230508.3
2023-05-30T11:20:57.6632308Z Included Software: https://github.com/actions/runner-images/blob/win22/20230508.3/images/win/Windows2022-Readme.md
2023-05-30T11:20:57.6632613Z Image Release: https://github.com/actions/runner-images/releases/tag/win22%2F20230508.3
2023-05-30T11:20:57.6632816Z ##[endgroup]
2023-05-30T11:20:57.6632968Z ##[group]Runner Image Provisioner
2023-05-30T11:20:57.6634177Z 2.0.171.1
2023-05-30T11:20:57.6634319Z ##[endgroup]
2023-05-30T11:20:57.6635250Z Current image version: '20230508.3'
2023-05-30T11:20:57.6641705Z Agent running as: 'XADMIN'
2023-05-30T11:20:57.6729518Z Prepare release directory.
2023-05-30T11:20:57.7112198Z Release folder: D:\a\*
```
I have to admit, I am a complete newbie in azure devops - but I can certainly say, that the build was running with the old version and the old download path until it wasn't available anymore.

**Comment by Nimtz89** [comment]
@cloud20x Can you send logging from your latest run, like in your initial post?

**Comment by cloud20x** [comment]
@Nimtz89 
Sure:
<img width="1534" alt="image" src="https://github.com/Winvision/vsts-tasks/assets/98384112/68c954b4-bc07-4489-a454-b9bbf8b455bc">
Thanks a lot for your support - it is highly appreciated! 👍

**Comment by Nimtz89** [comment]
@cloud20x Issue has something to do with the extension build, we'll try to fix it as soon as possible

**Comment by Nimtz89** [comment]
#39

**Comment by Nimtz89** [comment]
@cloud20x the migration part is working again in version 1.0.26, we're still working on the collectionmanagment

**Comment by cloud20x** [comment]
@Nimtz89 Thanks a lot for your efforts. 
I just tested it again, and in version 1.0.30 collectionmanagement is still not working.
```
2023-06-02T06:22:17.0821123Z ##[error]Unhandled: Cannot find module 'documentdb'
```

**Comment by cloud20x** [comment]
@Nimtz89
1.0.30 is still the current version (checked a couple of minutes ago).
Can you please inform us when you plan to release a fix for the collectionmanagement?
We are dependent on that fix in one of our projects and would need to organise an alternative solution if we can't expect a fix in the next days.
Thanks in advance for your feedback and your efforts.

**Comment by Nimtz89** [comment]
@cloud20x We have some trouble recreating the issue locally, but we'll try and fix the issue tomorrow.

**Comment by Nimtz89** [comment]
@cloud20x Thank you for your patience, we just release version 1.1.0, the collectionmanagment tasks has been fixed and tested on a cosmos database, can you try the updated version and confirm it's working? 
Thank you in advance

**Comment by cloud20x** [comment]
@Nimtz89 Thanks a lot. The job is already running and the collectionmanagement step did not throw an error 👍 
The task is still running, I will report back again, when the job has completed :-)

**Comment by cloud20x** [comment]
@Nimtz89 
The job is still running, but I received an error in one of the tasks
<img width="999" alt="image" src="https://github.com/Winvision/vsts-tasks/assets/98384112/269ef5be-ac86-4e4c-83ed-9790cdad9cee">

**Comment by cloud20x** [comment]
Hi @Nimtz89 
I run into errors like these:
```
2023-06-06T16:13:41.0827861Z D:\a\r1\a\myfile.json (1, 1431314): Message: {"Errors":["Request rate is large. More Request Units may be needed, so no changes were made. Please retry this request later. Learn more: http://aka.ms/cosmosdb-error-429"]}
2023-06-06T16:13:41.0828797Z ActivityId: xxx, Request URI: /apps/qqq/services/yyy/partitions/zzz/replicas/ttt/, RequestStats: 
2023-06-06T16:13:41.0832493Z RequestStartTime: 2023-06-06T16:02:21.8001278Z, Number of regions attempted: 1
2023-06-06T16:13:41.0833015Z , SDK: documentdb-dotnet-sdk/2.2.1 Host/64-bit MicrosoftWindowsNT/6.2.9200.0
2023-06-06T16:13:41.0833624Z D:\a\r1\a\myfile.json (1, 3700952): Message: {"Errors":["Request rate is large. More Request Units may be needed, so no changes were made. Please retry this request later. Learn more: http://aka.ms/cosmosdb-error-429"]}
2023-06-06T16:13:41.0834379Z ActivityId: xxx, Request URI: /apps/qqq/services/yyy/partitions/zzz/replicas/ttt/,  RequestStats: 
2023-06-06T16:13:41.0834884Z RequestStartTime: 2023-06-06T16:02:24.2903081Z, Number of regions attempted: 1
```

Thanks a lot for your support!

**Comment by Nimtz89** [comment]
@cloud20x What is the throughput (RU/s) on your collections and is the workload larger then it would normally be?

**Comment by cloud20x** [comment]
@Nimtz89 RUs was set to 4000 (as it used to be before, when the job was working). I used the same input file and the workload is "exactly" the same as before - no other processes running against the DB, when the job is running.
I tried to run the job with 10000 RUs, with the result that one of the tasks did not spit an error anymore, but the other one still did. But the job used to work with 1500 RUs as well. I don't think that we need to have 10000 RUs for our relatively small input files.
